src/silicone/utils.py

```diff
@@ -344,54 +352,46 @@ def convert_units_to_MtCO2_equiv(df, use_AR4_data=False):
         The input data with units converted.
     """
     # Check things need converting
-    if all(y[0:6] == "Mt CO2" for y in df.variables(True)["unit"]):
+    convert_to_str = "Mt CO2-equiv/yr"
+    if (df["unit"] == convert_to_str).all():
```

Comment:
You seem to have changed this code so it now changes "Mt CO2/yr" to "Mt CO2-equiv/yr". I don't know why you're doing this, I think it's confusing to put CO2 in CO2-equiv units.

Reply:
Ok I wasn't sure what behaviour you wanted. The options I saw were either a) if you use `convert_units_to_MtCO2_equiv`, then every unit in your dataframe is converted to 'Mt CO2-equiv/yr' (including CO2) or b) if you use `convert_units_to_MtCO2_equiv`, then every unit in your dataframe is converted to 'Mt CO2-equiv/yr' except for CO2 which is left as e.g. 'Mt CO2 / yr'. The reason I went for a is that I thought you'd want everything to be identical so that later operations (e.g. adding everything up) wouldn't get confused by the presence of two different units. I can switch if you would like.

As one other thought, I wouldn't use 'CO2-equiv' as the unit. Having a hyphen in the unit doesn't work with Pint so I would recommend a) simply using 'CO2' (in my opinion it's obvious that it's equivalent if the emissions themselves aren't CO2 e.g. if you have Emissions|CH4 in units of Mt CO2/yr then it's obvious that's an equivalent unit) or b) using 'CO2e' (like iiasa-energy-units, and I would just add that unit into openscm-units).

Reply:
My goal so far has been to ensure the output is consistent with the way the database currently works, which is "-equiv" after gas name if it's not that gas itself, and some indication of what equivalency metric is used in the variable name if you're really lucky. The benefit of having a long string like '-equiv' is that you can just strip it out before doing any operations (as I currently do), whereas if we just have 'e' I suspect that will go wrong very shortly. I guess "equiv" without a hyphen works and we just strip both out of everything (probably by some utils function in case this changes again), it just makes the database untidy. The results of this function usually won't propagate into reported data though, it's just for making an aggregated value.

Reply:
alright let's stick with being database consistent then. We can propagate a better solution through the tools later and just have some quick stripping tools for now.

Reply:
ammended in 9aaf02